cpp/src/fil/infer.cu

```diff
@@ -205,7 +205,8 @@ struct tree_aggregator_t {
     : tmp_storage(finalize_workspace) {}
 
   __device__ __forceinline__ void accumulate(
-    vec<NITEMS, float> single_tree_prediction, int tree) {
+    vec<NITEMS, float> single_tree_prediction, int tree, int num_rows) {
+    for (int i = num_rows; i < NITEMS; ++i) acc[i] = 0.0f;
```

Comment:
fixing a bug which somehow did not trigger the test in the previous PR. It triggers it now.